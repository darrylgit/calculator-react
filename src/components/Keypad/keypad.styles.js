import styled from 'styled-components';

export const KeypadContainer = styled.div`
  height: 65%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;

  align-items: center;
  align-content: space-around;

  @media only screen and (max-width: 23em) {
    height: 68%;
  }

  @media screen and (min-device-aspect-ratio: 1/1) and (orientation: landscape) and (hover: none) and (max-height: 40em) {
    height: 73%;
  }
`;
