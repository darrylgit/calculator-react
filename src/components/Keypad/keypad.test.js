import { shallow } from 'enzyme';
import React from 'react';

import Keypad from './keypad.component';

import {
  ADD,
  SUBTRACT,
  MULTIPLY,
  DIVIDE,
  CLEAR,
  BACKSPACE,
  PARENTHESES,
  NEGATIVE,
  EQUALS,
  DECIMAL
} from '../../constants';

describe('render keypad', () => {
  let keypad;
  beforeEach(() => {
    keypad = shallow(<Keypad />, {
      context: { values: [], pushValue: () => {} }
    });
  });

  afterEach(() => {
    keypad.unmount();
  });

  it('renders without crashing', () => {
    expect(keypad.debug()).toMatchSnapshot();
  });

  it('renders all the keys', () => {
    expect(keypad.find('InputKey').length).toEqual(20);

    const keyConstants = [
      CLEAR,
      BACKSPACE,
      PARENTHESES,
      ADD,
      DIVIDE,
      SUBTRACT,
      MULTIPLY,
      EQUALS,
      NEGATIVE,
      DECIMAL,
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9
    ];

    keyConstants.forEach(constant => {
      expect(
        keypad.find('InputKey').findWhere(n => n.prop('value') === constant)
          .length
      ).toEqual(1);
    });
  });
});
