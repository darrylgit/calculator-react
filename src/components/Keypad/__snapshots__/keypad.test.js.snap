// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render keypad renders without crashing 1`] = `
"<styled.div>
  <InputKey value=\\"CLEAR\\" label=\\"C\\" type=\\"clear\\" />
  <InputKey value=\\"BACKSPACE\\" label={{...}} type=\\"util\\" />
  <InputKey value=\\"PARENTHESES\\" label=\\"( )\\" type=\\"util\\" />
  <InputKey value=\\" ÷ \\" label=\\" ÷ \\" type=\\"operator\\" />
  <InputKey value=\\"7\\" label=\\"7\\" type=\\"num\\" />
  <InputKey value=\\"8\\" label=\\"8\\" type=\\"num\\" />
  <InputKey value=\\"9\\" label=\\"9\\" type=\\"num\\" />
  <InputKey value=\\" × \\" label=\\" × \\" type=\\"operator\\" />
  <InputKey value=\\"4\\" label=\\"4\\" type=\\"num\\" />
  <InputKey value=\\"5\\" label=\\"5\\" type=\\"num\\" />
  <InputKey value=\\"6\\" label=\\"6\\" type=\\"num\\" />
  <InputKey value=\\" − \\" label=\\" − \\" type=\\"operator\\" />
  <InputKey value=\\"1\\" label=\\"1\\" type=\\"num\\" />
  <InputKey value=\\"2\\" label=\\"2\\" type=\\"num\\" />
  <InputKey value=\\"3\\" label=\\"3\\" type=\\"num\\" />
  <InputKey value=\\" + \\" label=\\" + \\" type=\\"operator\\" />
  <InputKey value=\\"NEGATIVE\\" label={{...}} type=\\"util\\" />
  <InputKey value=\\"0\\" label=\\"0\\" type=\\"num\\" />
  <InputKey value=\\"DECIMAL\\" label=\\".\\" type=\\"util\\" />
  <InputKey value=\\"EQUALS\\" label=\\"=\\" type=\\"equals\\" />
</styled.div>"
`;
